ecs-agent/tcs/model/ecstcs/types.go

```diff
@@ -50,3 +52,84 @@ type HealthMessage struct {
 	Metadata      *HealthMetadata
 	HealthMetrics []*TaskHealth
 }
+
+// InstanceStatusMessage represents a message containing instance health status
+// information to be published to the TCS backend.
+type InstanceStatusMessage struct {
+	// Metadata contains identifying information about the container instance
+	// including cluster name, container instance ARN, and request ID.
+	Metadata *InstanceStatusMetadata `json:"metadata,omitempty"`
+
+	// Statuses contains a collection of instance status checks that represent
+	// the health state of various components on the container instance.
+	Statuses []*InstanceStatus `json:"statuses,omitempty"`
+}
+
+const (
+	InstanceHealthCheckTypeContainerRuntime = "ContainerRuntime"
+	InstanceHealthCheckTypeAgent            = "Agent"
+	InstanceHealthCheckTypeEBSDaemon        = "EBSDaemon"
+	InstanceHealthCheckTypeNvidia           = "NvidiaAcceleratedHardware"
+)
+
+const (
+	// HealthcheckStatusInitializing is the zero state of a healthcheck status.
+	InstanceHealthCheckStatusInitializing InstanceHealthCheckStatus = iota
```

Comment:
I know this pattern exists elsewhere in the codebase. But, how about using `string` constants directly instead of integers which then need to be converted to strings anyway? Using `iota` makes it hard to inject new statuses (that has bitten me in the past) between existing statuses. By using `string` type none of the translation functions would be required. 

Maybe like - 
```go
type InstanceHealthCheckStatus string

const (
    InstanceHealthCheckStatusInitializing InstanceHealthCheckStatus = "INITIALIZING"
    InstanceHealthCheckStatusOk          InstanceHealthCheckStatus = "OK"
    InstanceHealthCheckStatusImpaired    InstanceHealthCheckStatus = "IMPAIRED"
)
```